avm/src/daemon/daemon.rs

```diff
@@ -178,7 +192,92 @@ async fn run_agz_b64(agz_b64: &str) -> DaemonResult<()> {
   Ok(())
 }
 
+fn write_b64_file(file_name: &str, content: &str) -> DaemonResult<()> {
+  let pwd = env::current_dir();
+  match pwd {
+    Ok(pwd) => write(
+      format!("{}/{}", pwd.display(), file_name),
+      base64::decode(content).unwrap(),
+    )?,
+    Err(err) => {
+      let err = format!("{:?}", err);
+      return Err(err.into());
+    }
+  }
+  Ok(())
+}
+
+async fn get_files_b64() -> HashMap<String, String> {
+  let mut files_b64 = HashMap::new();
+  files_b64.insert("Dockerfile".to_string(), get_dockerfile_b64().await);
+  files_b64.insert("app.tar.gz".to_string(), get_app_tar_gz_b64().await);
+  files_b64
+}
+
+fn create_app_tar(content: Option<&String>) -> () {
+  if let Some(content) = content {
+    if let Err(err) = write_b64_file("app.tar.gz", &content) {
+      eprintln!("Could not create app.tar.gz. {:?}", err);
+      std::process::exit(1);
+    }
+  }
+}
+
+async fn generate_token() -> String {
+  let body = json!({
+    "deployConfig": deploy::get_config().await,
+  });
+  post_v1("localDaemonToken", body).await
+}
+
+async fn set_local_daemon_props() -> () {
+  let files_b64 = get_files_b64().await;
+  create_app_tar(files_b64.get(&"app.tar.gz".to_string()));
+  DAEMON_PROPS
+    .set(DaemonProperties {
+      clusterId: "daemon-local-cluster".to_string(),
+      agzB64: get_base_agz_b64(),
+      deployToken: generate_token().await,
+      domain: "alandeploy.com".to_string(),
+      filesB64: files_b64,
+    })
+    .unwrap();
+}
+
+fn create_certs_if_local() -> () {
+  if ALAN_TECH_ENV.as_str() == "local" {
+    // Self signed certs for local dev
+    // openssl req -newkey rsa:2048 -nodes -keyout key.pem -x509 -days 365 -out certificate.pem -subj "/C=US/ST=California/O=Alan Technologies, Inc/CN=*.anycloudapp.com"
+    let subj = if cfg!(target_os = "windows") {
+      "//C=US\\ST=California\\O=Alan Technologies, Inc\\CN=*.anycloudapp.com"
```

Comment:
Are you sure this is necessary? This isn't a filesystem path.

Reply:
hmm will check again, I do not remember why I've done it but remember to have read something about it.